src/output-json.c

```diff
@@ -437,8 +438,15 @@ void EvePacket(const Packet *p, JsonBuilder *js, unsigned long max_length)
         return;
     }
     if (!jb_set_uint(js, "linktype", p->datalink)) {
+        jb_close(js);
         return;
     }
+
+    const char *dl_name = LinktypeName(p->datalink);
+    // Intentionally ignore the return value from jb_set_string and proceed
+    // so the jb object is closed
+    jb_set_string(js, "linktype_name", dl_name == NULL ? "n/a" : dl_name);
```

Comment:
I would not log it if it is NULL

Reply:
like
```
if (dl_name) 
   jb_set_string(js, "linktype_name", dl_name);
```

We still have the integer value when this happens

Reply:
It can be nice to know that there will always be a value there.

Reply:
Yes.  But its open for discussion.  Then that name has to be something that can never conflict with a valid name though.